### PR TITLE
refactor: limited the usage of player input

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description

**Fixes**: # (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:

### Checklist

- [ ] My code follows the [Coding Conventions](#coding-conventions)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Runtime/Input/InputController.cs [hunk 1]
```diff
@@ -9,14 +10,25 @@ namespace Innoactive.Creator.Core.Input
...
+            {
+                Context = context;
+            }
+        }
+
         private struct ListenerInfo
         {
             public readonly IInputActionListener ActionListener;
-            public readonly Action<InputAction.CallbackContext> Action;
+            public readonly Action<InputEventArgs> Action;
```
Comment (Gusinuhe) [inline Runtime/Input/InputController.cs:29]: This is an API break, tbh there shouldn't be any real issue, but let's remember it.

**Review by Gusinuhe — APPROVED** [review]

File: Runtime/Input/DefaultInputController.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+using System;
+using System.Collections.Generic;
+using Innoactive.Creator.Core.Configuration;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Innoactive.Creator.Core.Input
+{
+    [RequireComponent(typeof(PlayerInput))]
+    public class DefaultInputController : InputController
```
Comment (tomwim) [inline Runtime/Input/DefaultInputController.cs:12]: Docs pls

File: Runtime/Input/InputController.cs [hunk 3]
```diff
@@ -33,6 +29,18 @@ public InputFocusEventArgs(IInputFocus inputFocus)
             }
         }
 
+        protected struct ListenerInfo
```
Comment (tomwim) [inline Runtime/Input/InputController.cs:32]: this also would like some docs please

**Review by tomwim — CHANGES_REQUESTED** [review]

File: Editor/Input/InputEditorUtils.cs [hunk 4]
```diff
@@ -35,6 +35,48 @@ public static void CopyCustomKeyBindingAsset()
         public static bool UsesCustomKeyBindingAsset()
         {
             return AssetDatabase.GetAssetPath(RuntimeConfigurator.Configuration.CurrentInputActionAsset).Equals("Assets/Resources" + RuntimeConfigurator.Configuration.CustomInputActionAssetPath);
+
```
Comment (Gusinuhe) [inline Editor/Input/InputEditorUtils.cs:45]:
```suggestion
```

File: Editor/Input/InputEditorUtils.cs [hunk 5]
```diff
@@ -35,6 +35,48 @@ public static void CopyCustomKeyBindingAsset()
...
+        /// </summary>
+        public static void OpenKeyBindingEditor()
+        {
+            if (UsesCustomKeyBindingAsset() == false)
+            {
+                CopyCustomKeyBindingAsset();
+            }
+
+            AssetDatabase.OpenAsset(RuntimeConfigurator.Configuration.CurrentInputActionAsset);
+
```
Comment (Gusinuhe) [inline Editor/Input/InputEditorUtils.cs:59]:
```suggestion
```

File: Editor/Input/InputEditorUtils.cs [hunk 6]
```diff
@@ -35,6 +35,48 @@ public static void CopyCustomKeyBindingAsset()
...
+        {
+            Debug.LogError("Error, no implementation for the old input system");
+        }
+
+        /// <summary>
+        /// Checks if the custom key bindings are loaded.
+        /// </summary>
+        public static bool UsesCustomKeyBindingAsset()
+        {
+
```
Comment (Gusinuhe) [inline Editor/Input/InputEditorUtils.cs:75]:
```suggestion
```

File: Editor/Input/InputEditorUtils.cs [hunk 7]
```diff
@@ -35,6 +35,48 @@ public static void CopyCustomKeyBindingAsset()
...
+            Debug.LogError("Error, no implementation for the old input system");
+            return false;
+        }
+
+        /// <summary>
+        /// Opens the key binding editor.
+        /// </summary>
+        public static void OpenKeyBindingEdit()
```
Comment (Gusinuhe) [inline Editor/Input/InputEditorUtils.cs]:
```suggestion
        public static void OpenKeyBindingEditor()
```

**Review by Gusinuhe — CHANGES_REQUESTED** [review]
Perhaps change the `Debug.Error` for `Exceptions` in `Editor/Input/InputEditorUtils.cs`

**Review by Gusinuhe — DISMISSED** [review]

File: Runtime/Input/DefaultInputController.cs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+using System;
+using System.Collections.Generic;
+using Innoactive.Creator.Core.Configuration;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Innoactive.Creator.Core.Input
+{
+    /// <summary>
+    /// Input controller based on unity's InputSystem.
```
Comment (tomwim) [inline Runtime/Input/DefaultInputController.cs:12]:
```suggestion
    /// Input controller based on Unity's InputSystem.
```

**Review by tomwim — DISMISSED** [review]

**Review by tomwim — APPROVED** [review]

**Pull request merged**